### PR TITLE
Add observer helper functions and binaries and hotfix binary for dht database migrations.

### PR DESCRIPTION
**Pull request by xq262144**
1. Add observer helper functions and binaries.
2. Add hotfix binary for dht database migrations.

### CODE REVIEW

**Review by zeqing-guo — APPROVED** [review]

File: cmd/hotfix/msgpack-20180824/main.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	dhtFile    string
+	privateKey string
+)
+
+func init() {
+	flag.StringVar(&dhtFile, "dhtFile", "dht.db", "dht database file to fix")
+	flag.StringVar(&privateKey, "private", "private.key", "private key to use for signing")
+}
+
+type OldBlock ct.Block
```
Comment (covenantdb) [inline cmd/hotfix/msgpack-20180824/main.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type OldBlock should have comment or be unexported

File: cmd/hotfix/msgpack-20180824/main.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+func (b *OldBlock) MarshalBinary() ([]byte, error) {
```
Comment (covenantdb) [inline cmd/hotfix/msgpack-20180824/main.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method OldBlock.MarshalBinary should have comment or be unexported

File: cmd/hotfix/msgpack-20180824/main.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	return nil, nil
+}
+
+func (b *OldBlock) UnmarshalBinary(data []byte) (err error) {
```
Comment (covenantdb) [inline cmd/hotfix/msgpack-20180824/main.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method OldBlock.UnmarshalBinary should have comment or be unexported

File: cmd/hotfix/msgpack-20180824/main.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		&b.SignedHeader.Signee,
+		&b.SignedHeader.Signature); err != nil {
+		return
+	}
+
+	err = ReadElements(reader, binary.BigEndian, &b.Queries)
+	return
+}
+
+type ServiceInstance struct {
```
Comment (covenantdb) [inline cmd/hotfix/msgpack-20180824/main.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type ServiceInstance should have comment or be unexported

**Pull request merged**